ufl/algorithms/apply_restrictions.py

```diff
@@ -129,11 +129,7 @@ def reference_value(self, o):
     def coefficient(self, o):
         "Allow coefficients to be unrestricted (apply default if so) if the values are fully continuous across the facet."
         e = o.ufl_element()
-        d = e.degree()
-        f = e.family()
-        # TODO: Move this choice to the element class?
-        continuous_families = ["Lagrange", "Q", "S"]
-        if (f in continuous_families and d > 0) or f == "Real":
+        if e.is_fully_continuous():
```

Comment:
Is this really the right fix? ufl already tracks the Sobolev space embedding of finite elements so I think this could be written `e in ufl.sobolevspace.H1` without adding a sui generis method to elements that only deals with the particular case of fully continuous.

Reply:
CR is in h1 (noncorming) according to ufl, but not fully continuous I think

Reply:
Looks to me like UFL has CR in L2. I'll try making this function check if the element is in H1 or H2 and see if all the tests still pass...

Reply:
The tests still pass 🥳